plugins/catalog-backend-module-ldap/src/ldap/read.ts

```diff
@@ -341,11 +351,23 @@ function mapReferencesAttr(
     const values = vendor.decodeStringAttribute(entry, attributeName);
     const dn = vendor.decodeStringAttribute(entry, vendor.dnAttributeName);
     if (values && dn && dn.length === 1) {
-      setter(dn[0], values);
+      if (vendor.dnCaseSensitive) {
+        setter(
+          dn[0].toLowerCase(),
+          values.map(v => v.toLowerCase()),
+        );
+      } else {
+        setter(dn[0], values);
+      }
     }
   }
 }
 
+/** Validates value exists and if required forced sensitivty value to lowercase */
+function getCaseSensitivityValue(value: string, vendor: LdapVendor) {
+  return value && vendor.dnCaseSensitive ? value.toLowerCase() : value;
```

Comment:
I'm not sure what's the best way to lowercase here, when taking different locales into account. Using `toLowerCase` will for example not translate uppercase i to i, but to a special i without a dot in Turkish locales which in some circumstances can be highly surprising. But doing `toLocaleLowerCase('en-US')` that we do in some other places will instead probably just do the wrong thing for values that are intentionally in non-ascii casing. Hm. Well the fields that we use this for should not be containing odd chars, to maybe `toLocaleLowerCase('en-US')` is the right call.